meetings/2022/notes-2022-08-22.md

```diff
@@ -0,0 +1,214 @@
+ 
+EAO: ECH, in Fluent, you can have different variants in a selection message. The syntax of Fluent doesn’t require an annotation for what type of value selection should be performed on. Only at runtime is that type is known.  At runtime, we wrap values as a NumberValue for plural selects.  The actual behavior ends up depending on the runtime value. In Fluent, it’s not possible to expect the input value be of a certain type, so we don’t know if the input is the number 1 or if it’s the string `”ONE”`.
+ 
+ECH: I see the problem, but it sounds like a Fluent problem.  We want to have a clear contract between the inputs of a formatter function and the output, and it relates to what I said earlier, which is that we select on formatted value.
+ 
+EAO: This is related to whether a user has the ability to pass options to a formatter.  What is the value of a value in a match statement. I think we should support partially formatted values in a select statement, and allow partially formatted values equal to other values.  It is hard for me to see how that would work if we always require functions for selectors.
+ 
+MIH: I want to refer back to a point from STA. He asked how is this (selectors) different from placeholders?  Most of the times, placeholders can be black boxes and they “just work” for formatting.  Selectors are not black boxes.  The tools won’t know what type of value the selector represents.  The idea that EAO was describing, that a formatted value would have methods on how to format itself, is 
+ 
+The argument of not supporting a one-way upgrade from Fluent messages to MFv2 is not a strong reason, and even by that measure, MessageFormat v1 is more established and widely used of a format so that should be more of a concern than Fluent for upgradeability, and we already acknowledged that it is a non-goal.
+ 
+STA: I acknowledge that we don’t want to put formatting functions in placeholders, that would be wasteful.. I think there is a point about consistency to be made. We all agree that we want to skip formatting functions in placeholders.  So if that is the case, then how is this case of selectors consistent with that of placeholders?
+ 
+EAO: I think there are 3 layers of type detection for linters, formatters, etc.  If we always have a function attached to a selector, then we don’t need to lookup what the type is. The second lookup is, say if we’re matching  `match {$foo} * {...}`, then we would want to specify in the syntax the types.  The third level is to look into the semantic comments in the message that declares that `$foo` is a number, which allows a linter / tooling to have access to that info.  My sense is that the place this becomes questionable, where the arguments are valid, are the questions about tooling and validation. We are likely to end up with more than one solution.  So maybe we end up with a selector without a function being a linting problem rather than a syntax problem.
```

Comment:
```suggestion
EAO: I think there are 3 layers of type detection for linters, formatters, etc.  If we always have a function attached to a selector, then we don’t need to lookup what the type is. The second lookup is, say if we’re matching  `match {$foo} * {...}`, then we could have `$foo` set by a local variable with a well-defined type.  The third level is to look into the semantic comments in the message that declares that `$foo` is a number, which allows a linter / tooling to have access to that info.  My sense is that the place this becomes questionable, where the arguments are valid, are the questions about tooling and validation. We are likely to end up with more than one solution.  So maybe we end up with a selector without a function being a linting problem rather than a syntax problem.
```